### ISSUE TITLE
Did you consider a multi lingual covid-at-home.github.io?!

### ISSUE THREAD

**Issue by blijblijblij** [covid-at-home/covid-at-home.github.io#7]
Someone has already that something like that for `jekyll` off course ;-)

https://github.com/sylvaindurand/jekyll-multilingual

and it would relief you from the hassle of starting new repo's based on the `en` version.

Just a thought, you might have already considered it, and the benefit of the current setup would be to leave it up to the localized teams to manage their own repo's eventually....

Just my two 💰

**Comment by ropg** [comment]
I must admit this is my first time doing anything with Jekyll, and with GitHub Pages, and with most other things. (Jumping in at the deep end is kind of normal for me.) I like the different language sites having their own domains, and the only easy way that doesn't involve anything running on other servers that I saw to do that is to make them separate HitHub organisations. Which is not that much of a hassle, and it indeed allows me to give the translators their own repository to work with.

Just this morning I wrote a script that spreads the centrally managed files (at the moment that's `_data/thankyou.yml` and `_data/languages.yml`) to all the different language sites from the english one.

What I could use is a script that creates a new language version and commits to the new repo, based on the english, using DeepL (if language available) or Google Translate. It would need to be moderately smart, probably Python or Ruby. Extra points for automating domain registration at joker.com, setting up the stats at goatcounter and creating GitHub org and repo first. Could do it but kinda busy now. Just in case you feel like spending a day. :)

**Comment by blijblijblij** [comment]
@ropg 👍 

I'll give it some thought this evening and see what I can do.

Fyi Github Actions https://github.com/covid-at-home/covid-at-home.github.io/actions/new is relatively new, projected as a CI / CD tool, but can be leveraged for pretty much any kind of automation. Might fit automating the distribution of those `_data/...` files.

**Comment by blijblijblij** [comment]
Doing some reading to see what can be done.

@ropg did you contact DeepL to get access to the DeepL api for this good cause for free?!

Api has some benefits like:
```
curl https://api.deepl.com/v2/translate \ 
	-d auth_key=your_auth_key \ 
	-d "text=This is the first sentence."  \ 
	-d "text=This is the second sentence."  \ 
	-d "text=This is the third sentence."  \ 
	-d "target_lang=DE"
```

But seems rather limited (`5 document translations per month`) when looking at https://www.deepl.com/pro.html#pricing

**Comment by ropg** [comment]
I got a DeepL business account, and because they limit the number of documents that can be uploaded, the current procedure (do not laugh...) is:

```
tar cvf text.tar covid-at-home.github.io/*.md
mv text.tar test.txt
```

Then upload this "text" file to deepl using web interface, download the translated copy, open it in BBEdit  and then manually copy the segments over the english version in BBEdit.

The first thing that makes this easier is a trivial file concatenator that just adds file name with a bunch of equal signs or so after it and then file contents so I don't have the skip the tar junk. Then a thing that would parse these files back out. And then something that does that and skips the liquid tags.

Google translate also mangles local links bc it doesn't recognize them, so they would also need to be skipped (and markdown links or image tags). Google Translate will do more languages than DeepL in the future bc the DeepL languages are almost exhausted: we have Russian, Chinese and Japanese left. So a file concatenator and a smart file splitter seem to make most sense and be most universal.

**Comment by ropg** [comment]
(And some way to get this into Google Translate, which may require splitting into <5000 char chunks instead of concatenating)

**Comment by blijblijblij** [comment]
First breakdown:

# covid-at-home-site-provisioner

Script for setting up a localised version of [covid-at-home.info](https://covid-at-home.info)

A script that

-   [ ] creates a new language version
-   [ ] commits to the new repo based on the english version
-   [ ] but with translated pages using [DeepL](https://www.deepl.com/translator) (if [language available](https://www.deepl.com/docs-api/other-functions/listing-supported-languages)) or [Google Translate](https://translate.google.com/)

Extra points for

-   [ ] automating domain registration at joker.com
-   [ ] setting up the stats at goatcounter and 
-   [ ] creating GitHub org and repo first

## Outline

    # English
    Org covid-at-home
    Repo covid-at-home.github.io
    Domain https://covid-at-home.info
      
    # Nederlands
    Org covid-thuis
    Repo covid-thuis.github.io
    Domain covid-thuis.info

    ...
    ...

    # $NEW_LANG_AT_HOME = "at home > translated to new language)
    Org covid-$NEW_LANG_AT_HOME
    Repo covid-$NEW_LANG_AT_HOME.github.io
    Domain covid-$NEW_LANG_AT_HOME.info

## Things to keep in mind

-   DeepL limit the number of documents that can be uploaded which makes for a [cumbersome procedure](https://github.com/covid-at-home/covid-at-home.github.io/issues/7#issuecomment-602035651) at this time, we might need to concatenate individual files to make this happen
-   While google translate [requires splitting texts into &lt;5000 char chunks](https://github.com/covid-at-home/covid-at-home.github.io/issues/7#issuecomment-602036742)

## Application (happy) flow

-   start cli and provide `--language`
-   google translate `at home` to `$NEW_LANG_AT_HOME`
-   check if `covid-$NEW_LANG_AT_HOME` can be used as `Org`
-   check if `covid-$NEW_LANG_AT_HOME.info` is available as domain
-   clone `en` version <https://github.com/covid-at-home/covid-at-home.github.io> without git object
-   concatenate `*.md` with suitable delimiter (so we can break them up at a later time) to one file

> if deepl supports the language
1.   deepl translate file in one go

> if deepl does not support the language
1.   break up one file in chunks &lt; 5000 char and let google translate do the smaller file
2.   combine the translated chunks back to one file


-   break up the one file based on delimiter
-   translate page title and save `en_title.md` as `translated_title.md`
-   create `covid-$NEW_LANG_AT_HOME.github.io` repo on github for `covid-$NEW_LANG_AT_HOME` Org
-   git init / commit the new folder
-   add remote
-   push
-   ... :money_with_wings:

**Comment by blijblijblij** [comment]
@ropg any recommendations on the model to use for the google `Cloud Translation API`?! 

Wanted to start by translating `at home` but the google api has a lot of options:

- AutoML Translation Model Training Operations
- AutoML Translation Online Predictions Operations
- Phrase-Based Translation Model Predictions
- Neural Translation Model Predictions
- Neural Translation Model Predictions In Translation V3

Some rather expensive, others for like 20 usd per million characters.

**Comment by blijblijblij** [comment]
Hmm, https://cloud.google.com/translate/docs/basic/setup-basic seems basic enough to start with

**Comment by ropg** [comment]
tl;dr ?

**Comment by ropg** [comment]
Wow, had only seen last mail...

Wow!

**Comment by ropg** [comment]
No idea about translations options. And I'm a bit fried right now: have been up since very early and not reading things right anymore all of a sudden. Shall we videoconference sometime tomorrow morning or whenever you're available? Just visit the [project videoconf](https://meet.jit.si/OngoingTeamMeetingForCovidAtHome), I'm often there.

**Comment by blijblijblij** [comment]
Sure, take care!

**Comment by blijblijblij** [comment]
@ropg Integrated with google translate and github (check org / repo) with some **WIP** over here 

https://github.com/blijblijblij/covid-at-home-site-provisioner/tree/task/Create-scaffold

Will swing by the video early evening as I'm a bit time limited during the day

**Comment by ropg** [comment]
Yo. That works well for me too...
